bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -220,6 +230,25 @@ impl ClientBuilder {
             );
         }
 
+        let mut certificates = Vec::new();
+
+        for certificate in builder.additional_root_certificates {
+            // We don't really know what type of certificate we may get here, so let's try
+            // first one type, then the other.
+            if let Ok(cert) = Certificate::from_der(&certificate) {
+                certificates.push(cert);
+            } else {
+                match Certificate::from_pem(&certificate) {
+                    Ok(cert) => {
+                        certificates.push(cert);
+                    }
+                    Err(e) => anyhow::bail!("Failed to add a root certificate {e:?}"),
+                }
```

Comment:
not sure if cuter or not, I'll let you decide :) 

```suggestion
                let cert = Certificate::from_pem(&certificate).unwrap_or_else(|e| anyhow::anyhow!("Failed to add a root certification {e}"))?;
                certificates.push(cert);
```